src/libraries/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Marshalling/ComImportInteropInterfaceDetailsStrategy.cs

```diff
@@ -0,0 +1,195 @@
+    /// The strategy provides <see cref="DynamicInterfaceCastableImplementationAttribute"/>-based implementations of <see cref="ComImportAttribute"/>-attributed interfaces by dynamically generating an interface using <see cref="System.Reflection.Emit"/> that has the following shape:
+    /// <code>
+    /// [assembly:IgnoresAccessChecksTo("AssemblyContainingIComInterface")]
+    /// [assembly:IgnoresAccessChecksTo("AssemblyContainingRetType")]
+    /// [assembly:IgnoresAccessChecksTo("AssemblyContainingArgType1")]
+    /// [assembly:IgnoresAccessChecksTo("AssemblyContainingArgType2")]
+    /// // One attribute per containing assembly of each type used in each method signature of the interface.
+    ///
+    /// namespace System.Runtime.CompilerServices
+    /// {
+    ///     [AssemblyUsage(AttributeTargets.Assembly, AllowMultiple = true)]
+    ///     internal class IgnoresAccessChecksToAttribute : Attribute
+    ///     {
+    ///         public IgnoresAccessChecksToAttribute(string assemblyName) { }
+    ///     }
+    /// }
+    ///
+    /// [DynamicInterfaceCastableImplementation]
+    /// interface InterfaceForwarder : IComInterface
+    /// {
+    ///     RetType IComInterface.Method1(ArgType1 arg1, ArgType2 arg2, ...)
+    ///     {
+    ///         return ((IComInterface)((IComImportAdapter)this).GetRuntimeCallableWrapper())(arg1, arg2, ...);
+    ///     }
+    /// }
+    /// </code>
+    ///
+    /// This mechanism allows source-generated COM interop to allow using runtime-implemented COM interfaces with runtime-defined marshalling behavior with minimal work on the source-generated COM interop side.
+    /// Additionally, by scoping the majority of the logic to this class, we make this logic more easily trimmable.
+    ///
+    /// We emit the <c>IgnoresAccessChecksToAttribute</c> to enable casting to internal <see cref="ComImportAttribute"/> types, which is a very common scenario (most <see cref="ComImportAttribute"/> types are internal).
+    /// </remarks>
+    [RequiresDynamicCode("Enabling interop between source-generated and runtime-generated COM requires dynamic code generation.")]
+    internal sealed class ComImportInteropInterfaceDetailsStrategy : IIUnknownInterfaceDetailsStrategy
+    {
+        public static readonly IIUnknownInterfaceDetailsStrategy Instance = new ComImportInteropInterfaceDetailsStrategy();
+
+        private readonly ConditionalWeakTable<Type, Type> _forwarderInterfaceCache = new();
+
+        // TODO: Support exposing ComImport interfaces through StrategyBasedComWrappers?
+        public IComExposedDetails? GetComExposedTypeDetails(RuntimeTypeHandle type) => DefaultIUnknownInterfaceDetailsStrategy.Instance.GetComExposedTypeDetails(type);
+
+        [UnconditionalSuppressMessage("Trimming", "IL2065", Justification = "Runtime-based COM interop is not supported with trimming enabled.")]
```

Comment:
Would it be better to mark the whole type as RequiresUnreferencedCode ?

Reply:
And place the warning suppression next to the existing one for AOT here: https://github.com/dotnet/runtime/pull/87583/files#diff-bc707b59bfbf9c250946780f9bdfb1e77347e5180bdf978e2247d49fc845a6b7R22 .